var/spack/repos/builtin/packages/py-pynucleus/package.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPynucleus(PythonPackage):
+    """PyNucleus is a finite element code that specifically targets nonlocal operators."""
+
+    homepage = "https://sandialabs.github.io/PyNucleus/index.html"
+    git = "https://github.com/sandialabs/PyNucleus.git"
+
+    maintainers("cgcgcg")
+
+    refs = ["master", "develop"]
+
+    for ref in refs:
+        version(ref, branch=ref)
+
+    depends_on("py-mpi4py@2.0.0:", type=("build", "link", "run"))
+    depends_on("py-cython", type="build")
+    depends_on("py-numpy", type=("build", "link", "run"))
+    depends_on("py-scipy", type=("build", "link", "run"))
+    depends_on("metis", type=("build", "link", "run"))
+    depends_on("parmetis", type=("build", "link", "run"))
+    depends_on("suite-sparse", type=("build", "link", "run"))
+    depends_on("py-h5py", type=("build", "link", "run"))
+    depends_on("py-tabulate", type=("build", "link", "run"))
+    depends_on("py-pyyaml", type=("build", "link", "run"))
+    depends_on("py-matplotlib", type=("build", "link", "run"))
+    depends_on("py-scikit-sparse", type=("build", "link", "run"))
+    depends_on("py-modepy", type=("build", "link", "run"))
+    depends_on("py-meshpy", type=("build", "link", "run"))
+    depends_on("py-pytools", type=("build", "link", "run"))
```

Comment:
I find it odd that most of these are build/link/run. Most Python dependencies are build/run only. Does the compiler actually link to these dependencies?